human-viz-bridge/src/iCubEffortBridge.cpp

```diff
@@ -0,0 +1,330 @@
+        robotTorques.zero();
+        
+        vector<string> jointSuffixes = {"_pitch", "_roll", "_yaw"};
+        for (size_t jointIndex = 0; jointIndex < robot_jointList.size(); jointIndex++) {
+            
+            string robotJoint = robot_jointList[jointIndex];
+            
+            size_t suffixStartingIndex = std::string::npos;
+            for (const std::string& suffix : jointSuffixes) {
+                suffixStartingIndex = robotJoint.find(suffix);
+                if (suffixStartingIndex != std::string::npos) {
+                    break;
+                }
+            }
+            
+            //two possible cases:
+            // - suffixStartingIndex != std::string::npos
+            // - suffixStartingIndex == std::string::npos
+            std::string mapKey = robotJoint;
+            if (suffixStartingIndex != std::string::npos) {
+                mapKey = mapKey.erase(suffixStartingIndex);
+            }
+            
+            map<std::string, EffortPublisher>::iterator effortPublisherFound = effortMap.find(mapKey);
+            if (effortPublisherFound == effortMap.end()) {
+                // create a new EffortPub
+                EffortPublisher newEffortPublisher;
+                // populate it with the correct fields
+                newEffortPublisher.linkEffort = robotModel.getLinkName(traversal.getChildLinkIndexFromJointIndex(robotModel, robotModel.getJointIndex(robotJoint)));
+                
+                // move here the new Publisher....
+                // insert it into the map,
+                std::pair<map<std::string, EffortPublisher>::iterator, bool> inserted = effortMap.insert(map<std::string, EffortPublisher>::value_type(mapKey, newEffortPublisher));
+                
+                //get back the iterator to this element
+                effortPublisherFound = inserted.first;
+                
+            }
+            
+            // do the "common" stuff
+            effortPublisherFound->second.jointDoFs.push_back(jointIndex);
+            cout << effortMap.find(mapKey)->first << " in the reference frame: " << effortMap.find(mapKey)->second.linkEffort << endl;
+             
+        }
+        
+        for (auto& effortList : effortMap) {
+            std::string topicName = rf.find("topicPrefix").asString() + "/" + effortList.first;
+            effortList.second.publisher = new Publisher<sensor_msgs_Temperature>();
+            if (!effortList.second.publisher) {
+                yError() << "Failed to create publisher to" << effortList.second.linkEffort;
+                return false;
+            }
+            if (!effortList.second.publisher->topic(topicName)) {
+                yError() << "Failed to create publisher to" << effortList.second.linkEffort;
+                return false;
+            }
+            
+        }
+        
+        for (auto& effortList : effortMap) {
+            string linkEffort; 
```

Comment:
You can probably move these lines inside the previous `for` given it iterates over the same elements